zingolib/src/lightclient.rs

```diff
@@ -946,16 +950,89 @@ impl LightClient {
             "outgoing_metadata" => outgoing_json,
         }
     }
-    /*
-        pub async fn do_list_txsummaries(&self) -> JsonValue {
-            for (txid, transaction_md) in self.wallet.transaction_context.transaction_metadata_set.read().await.current.iter()
-            {
-                let tx_value = transaction_md.total_value_spent();
-                let to_addresses =
-    }
-            Vec<TxSummary>
+    pub async fn do_list_txsummaries(&self) -> Vec<ValueTransferSummary> {
+        let mut sends = Vec::new();
+        let mut receipts = Vec::new();
+
+        for (txid, transaction_md) in self
+            .wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .current
+            .iter()
+        {
+            let tx_value_spent = transaction_md.total_value_spent();
+            let tx_value_received = transaction_md.total_value_received();
+            let tx_change_received = transaction_md.total_change_returned();
+            let TransactionMetadata {
+                block_height,
+                datetime,
+                zec_price,
+                ..
+            } = transaction_md;
+            match (tx_value_spent, (tx_value_received - tx_change_received)) {
+                //TODO: This is probably an error, if we sent no value and also received no value why
+                //do we have this transaction stored?
+                (0, 0) => unreachable!(),
+                // All received funds were change, this is a normal send
+                (_spent, 0) => {
+                    for OutgoingTxData {
+                        to_address,
+                        value,
+                        memo,
+                        recipient_ua,
+                    } in &transaction_md.outgoing_tx_data
+                    {
+                        if let Ok(to_address) = ZcashAddress::try_from_encoded(
+                            recipient_ua.as_ref().unwrap_or(to_address),
+                        ) {
+                            sends.push(ValueSendSummary {
+                                amount: *value,
+                                to_address,
+                                memo: memo.clone(),
+                                block_height: transaction_md.block_height,
```

Comment:
The `block_height`, `date_time`, and `price` are already recorded in the `TransactionMetadata`.